### ISSUE TITLE
Missing parameter in C++ methods

### ISSUE THREAD

**Issue by couet** [doxygen/doxygen#8620]
**Describe the bug**

In [our project ](https://root.cern/doc/master/index.html)we have code structures similar to:

File `AB.cxx`:

```
/// \class A
/// Class A
class A {
public:
   /// method 1 in A
   void method1(int i){}
};


/// \class B
/// Class B
class B : public A {
public:
   using A::method1;
   /// method 1 in B
   void method1(int i){}
   /// method 2
   void method2(int i){}
};
```

The following screenshot is what I get for "class B" . As you can see the parameter for `method1` defined in `class A` is missing in the detailed documentation.

**Expected behavior**

We would like to see the parameter for `method1()` in the class B detailed documentation.

**Screenshots**

<img width="1012" alt="Screenshot 2021-06-25 at 10 52 47" src="https://user-images.githubusercontent.com/4697738/123398722-252d2300-d5a4-11eb-892c-ad9dca3ed697.png">

**Version**
1.9.0 and 1.9.2

**Comment by albert-github** [comment]
I'm not sure whether the `method1` from `A` should be mentioned as a member function or as an inherited member, currently both mentions are present.

The fact that the parameter is missing in the detailed description is, as far as I can see, the result from the fact that the scope of using is not taken into account in the (function `MemberDefImpl::writeDocumentation` in file `memberdef.cpp`:
```
      const Definition *scope = cd;
      if (scope==0) scope = nd;
      hasParameterList=writeDefArgumentList(ol,scope,this);
```
for both `method1` the `scope->name()` is here `B` from the class `cd` but for the `method1` from class `A` the actual scope is probably `A` and is for that reason probably not matched.

@doxygen how to handle?

**Comment by doxygen** [comment]
In the example the `using A::method1` doesn't do anything it seems. Clients of class `B` can already invoke `B::method1` without the using statement, because the method is inherited from A. I can imagine this can be useful in case the method in class `A` is protected, and in class `B` one wants to expose it as a public method without adding a forwarding wrapper method. I think in this case, doxygen should show the method as a public method of class B. So what needs to be fixed is show the parameter list, as is requested.

**Comment by doxygen** [comment]
Look like a copy'n'paste mistake where the template argument list was assigned instead of the method's argument list 🙄. 
Please verify if the attached commit fixes the problem.

**Comment by albert-github** [comment]
From my tests it looks like the problem has been solved.

**Comment by couet** [comment]
I installed the master version to try if it was solved. But I get the following. The methods' helps are missing:

<img width="917" alt="Screenshot 2021-06-28 at 08 53 27" src="https://user-images.githubusercontent.com/4697738/123593187-83e5dd00-d7ee-11eb-8b05-792435423933.png">

**Comment by couet** [comment]
Indeed I see this is a new behaviour. The "Member function Documentation" is generated only if the help of the function contains more than just a "\brief" (or equivalent) . With the following code:
```
// \class A
/// Class A
class A {
public:
   /// method 1 in A
   void method1(int i){}
};


/// \class B
/// Class B
class B : public A {
public:
   using A::method1;
   /// method 1 in B.
   /// The is more help for method 1 in B.
   void method1(int i){}
   /// method 2 in B.
   /// The is more help for method 2 in B.
   void method2(int i){}
};
```
I get:

<img width="922" alt="Screenshot 2021-06-28 at 09 13 37" src="https://user-images.githubusercontent.com/4697738/123595704-80a02080-d7f1-11eb-82e7-31ab06676afd.png">

Which is now correct. We can wee the correct parameters' list.
Thanks for fixing it.

**Comment by doxygen** [comment]
@couet Alternatively, you can enable `ALWAYS_DETAILED_SEC` in the configuration file.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
